src/jdk.compiler/share/classes/com/sun/tools/javac/code/Symbol.java

```diff
@@ -2042,7 +2042,7 @@ public boolean binaryOverrides(Symbol _other, TypeSymbol origin, Types types) {
 
             // check for a direct implementation
             if (other.isOverridableIn((TypeSymbol)owner) &&
-                types.asSuper(owner.type, other.owner) != null &&
+                types.isSubtype(owner.type, other.owner.type) &&
```

Comment:
This seems inconsistent with what has been done below (see other comment)

Reply:
Good point.